### ISSUE TITLE
Wrong Solidity stacktrace line number

### ISSUE THREAD

**Issue by gorgos** [nomiclabs/buidler#482]
I just had a wrong line number given by buidler. As I am not sure what is causing it and it's a fairly complex system, I will just describe the error and given lines:

```
Error: VM Exception while processing transaction: revert ERC20: transfer amount exceeds allowance
      at TestToken.sub (@openzeppelin/contracts/math/SafeMath.sol:56)
      at TestToken.transferFrom (@openzeppelin/contracts/token/ERC20/ERC20.sol:100)
      at TestProtocol.safeTransferFrom (@openzeppelin/contracts/token/ERC20/SafeERC20.sol:25)
      at Protocol.deposit (contracts/impls/Protocol.sol:187)
      at Proxy.<fallback> (contracts/libs/upgrades/Proxy.sol:46)
      at TruffleContract.marginCallTrader (node_modules\@nomiclabs\truffle-contract\lib\execute.js:169:26)
      at Context.it.only (test\protocol.ts:282:24)
      at processTicksAndRejections (internal/process/task_queues.js:86:5)
```

- Incorrect given line Protocol.sol:187 = `IERC20(moneyMarket.baseToken()).safeTransferFrom(msg.sender, address(this), _baseTokenAmount);`
- Correct line Protocol.sol:318 = `IERC20(moneyMarket.baseToken()).safeTransferFrom(msg.sender, TRADER_MARGIN_CALL_FEE)`; (inside function `marginCallTrader `)
- The system is using proxy based upgradable contracts which may cause the issue.

**Comment by alcuadrado** [comment]
Hey @gorgos,

Do you have a link to reproduce this? Do you have the compiler optimizations enabled? The line numbers can be off with the optimizer.

**Comment by gorgos** [comment]
@alcuadrado That is my repo/config: https://github.com/laminar-protocol/flow-protocol-ethereum/blob/master/buidler.config.js. I have indeed the optimizer enabled.

`solc: { version: '0.6.4', optimizer: { enabled: true, runs: 200 } }`

**Comment by fzeoli** [comment]
Maybe Buidler EVM should warn the user if the optimizer is on

**Comment by gorgos** [comment]
A warning would be helpful. I needed the optimizer on, because I already hit the max gas limit of 9.5M, but my contract was too large without the optimizer. Allowing gas limits of greater than 9.5M would also be helpful.

**Comment by alcuadrado** [comment]
Hey @gorgos 

You can set the block gas limit as high as you want. Here's how: https://buidler.dev/config/#buidler-evm-network

You are probably hitting the EIP-170 contract size limit. It leads to an out of gas error. You can disable it in the newly released `1.3.0-rc.0` by setting `allowUnlimitedContractSize: true` in your Buidler EVM config.

**Comment by gorgos** [comment]
@alcuadrado I tried it today, but I can't get it to work. I am still seeing this message:

```
Error: Returned error: Transaction gas limit is 19500000 and exceeds block gas limit of 9500000
     at PromiEvent (node_modules\@nomiclabs\truffle-contract\lib\promievent.js:13:30)
      at C:\Users\mail\Personal\Laminar\flow-protocol-ethereum\node_modules\@nomiclabs\truffle-contract\lib\execute.js:223:26
      at Function.new (node_modules\@nomiclabs\truffle-contract\lib\contract\constructorMethods.js:57:53)
      at module.exports (test\truffle-fixture.js:11:44)
      at SimpleTaskDefinition.config_1.internalTask [as action] (node_modules\@nomiclabs\buidler-truffle5\src\index.ts:198:13)
```
